### PR TITLE
[ESP32] Update lighting-app with spake2p dependency for generation of factory partition

### PR DESCRIPTION
**Pull request by shubhamdp**
#### Problem
* Lighting app readme file do not contain step to build spake2p and add its path to $PATH environment variable

#### Change overview
* Updated readme

#### Testing
* Documentation update, no tests required

### CODE REVIEW

File: examples/lighting-app/esp32/README.md [hunk 1]
```diff
@@ -128,6 +128,25 @@ is "nvs".
 `scripts/tools/generate_esp32_chip_factory_bin.py` script generates the
 chip-factory NVS image `partition.bin`.
 
+Script has dependency on
```
Comment (dhrishi) [inline examples/lighting-app/esp32/README.md:131]: LGTM. Generic comment: Should we move the entire `Using ESP32 Factory Data Provider` section before `-   To build the demo application.`?

**Review by dhrishi — APPROVED** [review]

File: examples/lighting-app/esp32/README.md (hunk 1 above)
Comment (shubhamdp) [reply examples/lighting-app/esp32/README.md]: Moved to "Building the Example Application"

**Review by bzbarsky-apple — APPROVED** [review]

**Review by isiu-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking to bypass review, I filed an issue against PullApprove

**Pull request merged**